packages/core/src/i18n/language-detector.js

```diff
@@ -0,0 +1,9 @@
+export default {
+  name: 'mainTag',
+
+  lookup(options) {
+    const mainTag = document.querySelector('main');
+
+    return mainTag?.getAttribute('lang');
+  },
+};
```

Comment:
This, along with the `languageDetector.addDetector(mainTag);` in the setup file, allows the language detection to look at the `lang` attribute of the `<main>` tag. It's modeled after the dependency's [built-in `htmlTag` detector](https://github.com/i18next/i18next-browser-languageDetector/blob/34e9180e753ea3343b0ec5d079dd36fe46851566/src/browserLookups/htmlTag.js#L1-L14).